docs: blurb about msq union all (#15223)
apache/druid: docs/querying/sql.md

SELECT COUNT(*) FROM tbl WHERE my_column = 'value2'
```

Keep the following in mind when writing top-level UNION ALL queries:

> - You can't apply GROUP BY, ORDER BY, or any other operator to the results of a UNION ALL.

Reviewer: Can coalesce into a single paragraph as mentioned in the other comment
```suggestion

```

Author: This limitation predates my changes. So does it no longer apply to non-MSQ queries?

Reviewer: This limitation is still present, but as per the previous comment, we can make it cohesive by coalescing both of the limitations into one big para.

Author: Ok, if we want it in 1 paragraph, I'll need to rewrite the paragraph first. Starting with `If you use the MSQ task engine for the query,` sets the reader up to think that the whole paragraph is only about the MSQ task engine limitations

Author: https://github.com/apache/druid/pull/15223/commits/9ab413099ce3a7a68f42cd75ba8b08ad29d702c7 has the new paragraph

> - If you use the MSQ task engine for the query, the SQL planner attempts to plan the top-level UNION ALL as a table-level UNION ALL. Because of this, the query behaves the same as a table-level UNION ALL, meaning it has the same characteristics and limitations. If the planner can't plan the query as a table-level UNION ALL, the query fails.

Reviewer: ```suggestion
- If you use the MSQ task engine for the query, the SQL planner attempts to plan the top-level UNION ALL as a table-level UNION ALL. Therefore it has the same limitations as the table level union all - You can only union TableDataSources with simple projections - select columns and rename columns.  Clauses like GROUP BY, ORDER BY, etc are not supported on operands of union. 
The query isn't supported if the planner can't plan it as a table-level union all
```

SELECT COUNT(*) FROM tbl WHERE my_column = 'value2'
```

> When you use a top-level UNION ALL, there are certain limitations that apply. For all top-level UNION ALL queries, you can't apply a GROUP BY, ORDER BY, or any other operator to the results of the query. For any top-level UNION ALL that uses the MSQ task engine, the SQL planner attempts to plan the top-level UNION ALL as a table-level UNION ALL. Because of this, UNION ALL queries that use the MSQ task engine always behave the same as a table-level UNION ALL queries. They have the same characteristics and limitations. If the planner can't plan the query as a table-level UNION ALL, the query fails.

Reviewer 2: ```suggestion
Certain limitations apply when you use a top-level UNION ALL. For all top-level UNION ALL queries, you can't apply a GROUP BY, ORDER BY, or any other operator to the results of the query. For any top-level UNION ALL that uses the MSQ task engine, the SQL planner attempts to plan the top-level UNION ALL as a table-level UNION ALL. Because of this, UNION ALL queries that use the MSQ task engine always behave the same as table-level UNION ALL queries. They have the same characteristics and limitations. If the planner can't plan the query as a table-level UNION ALL, the query fails.
```
